commandline/src/main/java/tech/pegasys/web3signer/commandline/subcommands/Eth2WatermarkRepairSubCommand.java

```diff
@@ -45,40 +49,75 @@ public class Eth2WatermarkRepairSubCommand implements Runnable {
       arity = "1")
   Long epoch;
 
   @Option(
       names = "--slot",
-      paramLabel = "<epoch>",
-      description = "Low watermark to set the block slot to.",
+      paramLabel = "<slot>",
+      description =
+          "Low watermark to set the block slot to. (Sets high watermark slot when --set-high-watermark=true).",
       arity = "1")
   Long slot;
 
+  @Option(
+      names = "--set-high-watermark",
+      paramLabel = "<BOOL>",
+      description =
+          "Sets high watermark to given epoch and slot. (Sets low watermark when --set-high-watermark=false)."
+              + " (Default: ${DEFAULT-VALUE})")
+  boolean setHighWatermark = false;
+
+  @Option(
+      names = "--remove-high-watermark",
+      paramLabel = "<BOOL>",
+      description =
+          "Removes high watermark. When set to true, all other subcommand options are ignored."
+              + " (Default: ${DEFAULT-VALUE})")
+  boolean removeHighWatermark = false;
+
   @Override
   public void run() {
     final LowWatermarkDao lowWatermarkDao = new LowWatermarkDao();
+    final MetadataDao metadataDao = new MetadataDao();
     final ValidatorsDao validatorsDao = new ValidatorsDao();
 
     final SlashingProtectionContext slashingProtectionContext =
         SlashingProtectionContextFactory.create(eth2Config.getSlashingProtectionParameters());
     final Jdbi jdbi = slashingProtectionContext.getSlashingProtectionJdbi();
 
-    final List<Validator> allValidators =
-        jdbi.inTransaction(h -> validatorsDao.findAllValidators(h).collect(Collectors.toList()));
-
-    allValidators.stream()
-        .parallel()
-        .forEach(
-            validator ->
-                jdbi.useTransaction(
-                    h -> {
-                      lowWatermarkDao.updateSlotWatermarkFor(
-                          h, validator.getId(), UInt64.valueOf(slot));
-                      lowWatermarkDao.updateEpochWatermarksFor(
-                          h, validator.getId(), UInt64.valueOf(epoch), UInt64.valueOf(epoch));
-                    }));
-    LOG.info("Updated low watermark for {} validators", allValidators.size());
+    if (removeHighWatermark) {
+      jdbi.useTransaction(metadataDao::deleteHighWatermark);
+      LOG.info("Removed high watermark for all validators");
+    } else if (setHighWatermark) {
+      jdbi.useTransaction(h -> setHighWatermark(h, metadataDao));
+      LOG.info("Updated high watermark for all validators");
+    } else {
+      final List<Validator> allValidators =
+          jdbi.inTransaction(h -> validatorsDao.findAllValidators(h).collect(Collectors.toList()));
+
+      LOG.info("Updating low watermark for all {} validators...", allValidators.size());
+
+      allValidators.stream()
+          .parallel()
+          .forEach(
+              validator ->
+                  jdbi.useTransaction(h -> setLowWatermark(h, validator, lowWatermarkDao)));
```

Comment:
Note, if the low watermark conflicts with the high watermark, this fails fast rather than the slashing import where it fails per validator.

Since we're updating all validators to the same low watermark, they should all fail anyway so this is fine.